server/test/unit/server/controllers/test_repository.py

```diff
@@ -223,138 +223,79 @@ def test_skip(self, mock_get_model, mock_demo_objects, mock_rcu_objects):
-            downloaded=False)
-        self.assertTrue(result)
-
-    @patch(MODULE + '.plugin_api.get_unit_model_by_id')
-    @patch(MODULE + '.model.RepositoryContentUnit')
-    def test_not_all_downloaded(self, association, get_model):
-        associations = [
-            Mock(unit_id=1, unit_type_id='dog'),
-            Mock(unit_id=2, unit_type_id='dog'),
-            Mock(unit_id=3, unit_type_id='dog'),
-            Mock(unit_id=4, unit_type_id='cat'),
-            Mock(unit_id=5, unit_type_id='cat'),
-            Mock(unit_id=6, unit_type_id='cat'),
-        ]
-        associations_q_set = Mock()
-        associations_q_set.only.return_value = associations
-        association.objects.return_value = associations_q_set
-        dog = Mock()
-        dog.objects.return_value.count.return_value = 0
-        cat = Mock()
-        cat.objects.return_value.count.return_value = 3
-        types = {
-            'dog': dog,
-            'cat': cat
-        }
-        get_model.side_effect = lambda t: types[t]
-        repository = Mock(repo_id='1234')
-
-        # test
-        result = repo_controller.has_all_units_downloaded(repository)
-
-        # validation
-        association.objects.assert_called_once_with(repo_id=repository.repo_id)
-        associations_q_set.only.assert_called_once_with('unit_id', 'unit_type_id')
-        dog.objects.assert_called_once_with(
-            id__in=tuple([a.unit_id for a in associations[0:3]]),
-            downloaded=False)
-        cat.objects.assert_called_once_with(
-            id__in=tuple([a.unit_id for a in associations[3:6]]),
-            downloaded=False)
-        self.assertFalse(result)
+class FindUnitsNotDownloadedTests(unittest.TestCase):
+
+    @patch(MODULE + '.get_mongoengine_repo_querysets')
+    def test_call(self, mock_repo_querysets):
+        repo = Mock(repo_id='mock_repo')
+        mock_qs = Mock()
+        mock_qs.return_value = [0, 1, 2]
+        mock_repo_querysets.return_value = [mock_qs]
+        units = repo_controller.find_units_not_downloaded(repo)
+        for x, unit in zip(range(3), units):
+            self.assertEqual(x, unit)
+
+
+class MissingUnitCountTests(unittest.TestCase):
+
+    @patch(MODULE + '.get_mongoengine_repo_querysets')
+    def test_call(self, mock_repo_querysets):
+        repo = Mock(repo_id='mock_repo')
+        query_set = Mock()
+        query_set.return_value.count.return_value = 5
+        mock_repo_querysets.return_value = [query_set]
+        self.assertEqual(5, repo_controller.missing_unit_count(repo))
+
+
+class HasAllUnitsDownloadedTests(unittest.TestCase):
+
+    @patch(MODULE + '.get_mongoengine_repo_querysets')
+    def test_true(self, mock_repo_querysets):
+        repo = Mock(repo_id='mock_repo')
+        query_set = Mock()
+        query_set.return_value.count.return_value = 0
+        mock_repo_querysets.return_value = [query_set]
+        self.assertTrue(repo_controller.has_all_units_downloaded(repo))
```

Comment:
The call being tested takes a repo_id now, not a repo object. The same problem appears in other nearby tests.